nibabel/nifti1.py

```diff
@@ -380,6 +380,113 @@ def write_to(self, fileobj, byteswap):
         # next 16 byte border
         fileobj.write(b'\x00' * (extstart + rawsize - fileobj.tell()))
 
+class Nifti1DicomExtension(Nifti1Extension):
+    """Class for NIfTI1 DICOM header extension.
+
+    This class is a thin wrapper around pydicom to read a binary DICOM
+    byte string. If pydicom is not available, it silently falls back to the
+    standard NiftiExtension class.
+    """
+    def __init__(self, code, content):
+        self._code = code
+        self._raw_content = content
+        self._bio = BytesIO(content)
+        self._content = self._unmangle(content)
+
+    def _check_encoding(self):
+        """DICOM Data can be stored in the header either as a valid DICOM
+        object, with a preamble, meta info, transfer syntax etc., or as a
+        set of naked tags.  Check for meta info and transfer synatx here
+        and fall back to heuristics if metainfo is missing."""
+        self._bio.seek(0)
+        self._preamble = read_preamble(self._bio,True)  # Attempt to read preamble,
+                                                        # skip if missing w/o error
+
+        if self._preamble:
+            self._meta,_is_implicit_VR,_is_little_endian = self._check_meta()
+        else:
+            self._meta = None
+            _is_implicit_VR = self._guess_implicit_VR()
+            _is_little_endian = self._guess_little_endian()
+        return _is_implicit_VR,_is_little_endian
+
+
+    def _check_meta(self):
+        """Check the DICOM Transfer Syntax and set encoding appropriately.
+        Extracted from dicom.filereader.read_partial, see there for detail"""
+        file_meta_dataset = _read_file_meta_info(self._bio)
+        transfer_syntax = file_meta_dataset.TransferSyntaxUID
+        if transfer_syntax == dicom.UID.ImplicitVRLittleEndian:
+            is_implicit_VR = True
+            is_little_endian = True
+        elif transfer_syntax == dicom.UID.ExplicitVRLittleEndian:
+            is_implicit_VR = False
+            is_little_endian = True
+        elif transfer_syntax == dicom.UID.ExplicitVRBigEndian:
+            is_implicit_VR = False
+            is_little_endian = False
+        elif transfer_syntax == dicom.UID.DeflatedExplicitVRLittleEndian:
+            zipped = fileobj.read()
```

Comment:
You're right - I took this directly from pydicom, although it wasn't abstracted in a method I could call from nibabel. The only transfer syntax tested here is ImplicitVRLittleEndian, not the zipped or big endian ones. I'll add tests and fix this.

I haven't been using Pyflakes, but just grabbed a bundle to enable it. Thanks for the suggestion!